pkg/operator/ceph/object/user/controller_test.go

```diff
@@ -353,3 +366,113 @@ func TestBuildUpdateStatusInfo(t *testing.T) {
+					return &http.Response{
+						StatusCode: 200,
+						Body:       ioutil.NopCloser(bytes.NewReader([]byte(userCreateJSON))),
+					}, nil
+				}
+			}
+
+			if req.Method == http.MethodPut {
+				if req.URL.RawQuery == "enabled=true&format=json&max-objects=10000&max-size=-1&quota=&quota-type=user&uid=my-user" ||
+					req.URL.RawQuery == "enabled=true&format=json&max-objects=0&max-size=10000000000&quota=&quota-type=user&uid=my-user" ||
+					req.URL.RawQuery == "enabled=true&format=json&max-objects=10000&max-size=10000000000&quota=&quota-type=user&uid=my-user" {
+					return &http.Response{
+						StatusCode: 200,
+						Body:       ioutil.NopCloser(bytes.NewReader([]byte(userCreateJSON))),
+					}, nil
+				}
+			}
+
+			return nil, fmt.Errorf("unexpected request: %q. method %q. path %q", req.URL.RawQuery, req.Method, req.URL.Path)
+		},
+	}
+	adminClient, err := admin.New("rook-ceph-rgw-my-store.mycluster.svc", "53S6B9S809NUP19IJ2K3", "1bXPegzsGClvoGAiJdHQD1uOW2sQBLAZM9j9VtXR", mockClient)
+	assert.NoError(t, err)
+	userConfig := generateUserConfig(objectUser)
+	r := &ReconcileObjectStoreUser{
+		objContext: &cephobject.AdminOpsContext{
+			AdminOpsClient: adminClient,
+		},
+		userConfig: &userConfig,
+	}
+
+	// Test 1: Quota and Cap is nil
+	err = r.createorUpdateCephUser(objectUser)
+	assert.NoError(t, err)
+
+	// Test 2: Setting MaxBuckets
+	objectUser.Spec.Quotas = &cephv1.RGWUserQuotaSpec{MaxBuckets: maxbucket}
+	userConfig = generateUserConfig(objectUser)
+	r.userConfig = &userConfig
+	err = r.createorUpdateCephUser(objectUser)
+	assert.NoError(t, err)
+
+	// Test 3: Setting Caps
+	objectUser.Spec.Quotas = nil
+	objectUser.Spec.Capabilities = &cephv1.CapSpec{
+		User:   "read",
+		Bucket: "read",
+	}
+	userConfig = generateUserConfig(objectUser)
+	r.userConfig = &userConfig
+	err = r.createorUpdateCephUser(objectUser)
+	assert.NoError(t, err)
+
+	// Test 4: Setting MaxObjects
+	objectUser.Spec.Capabilities = nil
+	objectUser.Spec.Quotas = &cephv1.RGWUserQuotaSpec{MaxObjects: maxobject}
+	userConfig = generateUserConfig(objectUser)
+	r.userConfig = &userConfig
+	err = r.createorUpdateCephUser(objectUser)
+	assert.NoError(t, err)
+
+	// Test 5: Setting MaxObjects
+	maxsize, err := resource.ParseQuantity(maxsizestr)
+	assert.NoError(t, err)
+	objectUser.Spec.Quotas = &cephv1.RGWUserQuotaSpec{MaxSize: &maxsize}
+	userConfig = generateUserConfig(objectUser)
+	r.userConfig = &userConfig
+	err = r.createorUpdateCephUser(objectUser)
+	assert.NoError(t, err)
+
+	// Test 6: Setting both Quotas and Caps
+	objectUser.Spec.Capabilities = &cephv1.CapSpec{
+		User:   "read",
+		Bucket: "read",
+	}
+	objectUser.Spec.Quotas.MaxObjects = maxobject
+	objectUser.Spec.Quotas.MaxBuckets = maxbucket
+	userConfig = generateUserConfig(objectUser)
+	r.userConfig = &userConfig
+	err = r.createorUpdateCephUser(objectUser)
+	assert.NoError(t, err)
```

Comment:
What are cases where we would expect this to return an error? There should be a test for each error we might get as well.